server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MimeMessageBenchTest.java

```diff
@@ -0,0 +1,146 @@
+        dataProbe.addUser("rcpt4@" + DEFAULT_DOMAIN, PASSWORD);
+        dataProbe.addUser("rcpt5@" + DEFAULT_DOMAIN, PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void receivedMessagesShouldContainDeliveredToHeaders() throws Exception {
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort());
+
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        IntStream.range(0, 100)
+            .forEach(Throwing.intConsumer(i -> messageSender.sendMessage(MailImpl.builder()
+                .name("name" + i)
+                .sender(FROM)
+                .addRecipients("rcpt1@" + DEFAULT_DOMAIN,
+                    "rcpt2@" + DEFAULT_DOMAIN,
+                    "rcpt3@" + DEFAULT_DOMAIN,
+                    "rcpt4@" + DEFAULT_DOMAIN,
+                    "rcpt5@" + DEFAULT_DOMAIN)
+                .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                    .setSubject("subject i")
+                    .setText(CONTENT))
+                .build())));
+
+        awaitAtMostOneMinute.until(() -> jamesServer.getProbe(SpoolerProbe.class).processingFinished());
+        System.out.println("Spent: " + stopwatch.elapsed(TimeUnit.MILLISECONDS));
+    }
+
+
+    private MailetContainer.Builder generateMailetContainerConfiguration() {
+        return TemporaryJamesServer.defaultMailetContainerConfiguration()
+            .putProcessor(ProcessorConfiguration.transport()
+                .addMailet(MailetConfiguration.builder()
+                    .matcher(RecipientIs.class)
+                    .matcherCondition("rcpt1@" + DEFAULT_DOMAIN)
+                    .mailet(LogMessage.class)
+                    .addProperty("passThrough", "true"))
+                .addMailet(MailetConfiguration.builder()
+                    .matcher(RecipientIs.class)
+                    .matcherCondition("rcpt2@" + DEFAULT_DOMAIN)
+                    .mailet(LogMessage.class)
+                    .addProperty("passThrough", "true"))
+                .addMailet(MailetConfiguration.builder()
+                    .matcher(RecipientIs.class)
+                    .matcherCondition("rcpt3@" + DEFAULT_DOMAIN)
+                    .mailet(LogMessage.class)
+                    .addProperty("passThrough", "true"))
+                .addMailet(MailetConfiguration.builder()
+                    .matcher(RecipientIs.class)
+                    .matcherCondition("rcpt4@" + DEFAULT_DOMAIN)
+                    .mailet(LogMessage.class)
+                    .addProperty("passThrough", "true"))
+                .addMailet(MailetConfiguration.builder()
+                    .matcher(RecipientIs.class)
+                    .matcherCondition("rcpt5@" + DEFAULT_DOMAIN)
+                    .mailet(LogMessage.class)
+                    .addProperty("passThrough", "true"))
+                .addMailetsFrom(CommonProcessors.deliverOnlyTransport()));
```

Comment:
while this configuration does split message, as far as I can tell, none of the mailets attempt to concurrenlty "write" to the message, hence the exercised concurrency is actually pretty low.

Reply:
If "writes" are performed, then content are not shareable anymore, and copy will happen anyway with the cow proxy.

Doing only reads, where things can be shared, and avoiding copy, was my intention...

Reply:
alright